app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -123,7 +123,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
         switch (requestCode) {
             case CommCareHomeActivity.UPGRADE_APP:
                 if (resultCode == RESULT_CANCELED) {
-                    Toast.makeText(this, R.string.update_canceled, Toast.LENGTH_LONG).show();
+                    UpdateTask task = UpdateTask.getRunningInstance();
```

Comment:
@amstone326 requires users to logout of their commcare session before updating. Not sure if that is necessary anymore, but that saves us from having to worry about multiple instances of UpdateTask floating around, which sounds great because I designed it as a singleton task.